smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/integrations/javadoc/JavadocInjectorInterceptor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.traitcodegen.integrations.javadoc;
+
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.traits.DeprecatedTrait;
+import software.amazon.smithy.model.traits.UnstableTrait;
+import software.amazon.smithy.traitcodegen.sections.ClassSection;
+import software.amazon.smithy.traitcodegen.sections.EnumVariantSection;
+import software.amazon.smithy.traitcodegen.sections.GetterSection;
+import software.amazon.smithy.traitcodegen.sections.JavaDocSection;
+import software.amazon.smithy.traitcodegen.writer.TraitCodegenWriter;
+import software.amazon.smithy.utils.CodeInterceptor;
+import software.amazon.smithy.utils.CodeSection;
+import software.amazon.smithy.utils.SmithyUnstableApi;
+
+/**
+ * Adds a javadoc section to classes, getters, and enum variants.
+ *
+ * <p>The injected javadoc section does not contain any javadoc contents when it is first injected. This
+ * section is instead populated by other {@code CodeInterceptors} that trigger off of specific traits.
+ * Note: This interceptor will also add any relevant documentation annotations to classes, getters, or enum variants.
+ */
+public class JavadocInjectorInterceptor implements CodeInterceptor.Prepender<CodeSection, TraitCodegenWriter> {
```

Comment:
```suggestion
public class JavaDocInjectorInterceptor implements CodeInterceptor.Prepender<CodeSection, TraitCodegenWriter> {
```

+ related changes for consistency.